packages/next/build/babel/preset.ts

```diff
@@ -74,7 +80,7 @@ module.exports = (context: any, options: NextBabelPresetOptions = {}): BabelPres
         corejs: 2,
         helpers: true,
         regenerator: true,
-        useESModules: !isTest && presetEnvConfig.modules !== 'commonjs',
+        useESModules: supportsESM && presetEnvConfig.modules !== 'commonjs',
```

Comment:
Would it be worth exposing an option to disable this check for users? Since people could still manually enable the commonjs plugin directly, this `presetEnvConfig.modules !== 'commonjs'` check isn't guaranteed to catch all cases where CommonJS is used.

Reply:
Users can already override the options because of `...options['transform-runtime']`

Reply:
Ahhh good call.